packages/core/src/ButtonBase/ButtonBase.js

```diff
@@ -0,0 +1,69 @@
+import React from 'react'
+import cx from 'classnames'
+
+import propTypes from '@dhis2/prop-types'
+
+import styles from './ButtonBase.styles.js'
+
+/**
+ * @module
+ * @param {ButtonBase.PropTypes} props
+ * @returns {React.Component}
+ */
+function ButtonBase(
+    {
+        children,
+        className,
+        primary,
+        secondary,
+        destructive,
+        small,
+        large,
+        disabled,
+        ...props
+    },
+    ref
+) {
+    return (
+        <button
+            {...props}
+            className={cx(className, {
+                primary,
+                secondary,
+                destructive,
+                small,
+                large,
+            })}
+            disabled={disabled}
+            ref={ref}
+        >
+            {children}
+            <style jsx>{styles}</style>
+        </button>
+    )
+}
+
+export default React.forwardRef(ButtonBase)
```

Comment:
I like the default exports, because then users can use a name that's most appropriate for them. Also great to have refs here 👍 

Why don't we just export from the bottom of the file though? That seems more conventional...

Reply:
It turned out that way for a few arbitrary reasons, so I can move it:

- The component needs to be a named function otherwise the name doesn't show up in devtools.
- We could wrap the function definition with the `React.forwardRef(function ButtonBase) {})` but I found it unclear.
- The `forwardRef` is important, so I wanted it as close to the function definition as possible.

Reply:
9711ab34